### DISCUSSION TITLE
State of cpp.js for React Native

### DISCUSSION THREAD

**Discussion by JakeStoeffler** [bugra9/cpp.js#33] [General]
Hi @bugra9 ! This is a really cool project, and I am particularly interested in using it for a new React Native app in order to leverage a lot of existing C++ code at our company. Could you provide me a general idea of where this library is at in terms of readiness for external use? I couldn't find much documentation on the React Native usage. Thanks in advance!

**Comment by bugra9** [comment]
Hi @JakeStoeffler,

Thanks for your interest! I'm really excited that you're considering using this project for your React Native app, especially to leverage your existing C++ code.

In terms of readiness, I'm planning to release a version that's ready for external use in about two weeks. I'll also be completed the documentation by then, so it should be easier for you to get started. Right now, I'm testing things by migrating the gdal3.js project to the cpp.js infrastructure and aiming to launch its mobile version soon. If you want to take a look, you can follow the progress [here](https://github.com/bugra9/gdal3.js/tree/v3.0.0-cpp.js-integration/apps/app-mobile), which might also give you some ideas for mobile usage.

I’m also hosting the dependencies for the gdal3.js project [here](https://github.com/bugra9/cpp.js/tree/main/packages). This should help you see how to package any C++ library into a cpp.js package. The packages are compiled for Android, iOS, and Emscripten platforms, so they should cover a wide range of use cases.

The bridges in the project are designed to work with modern C++. If the methods you need aren't implemented this way, you might need to write some wrappers. You can check out an example [here](https://github.com/bugra9/gdal3.js/tree/v3.0.0-cpp.js-integration/src/native).
